graphics/tinygl/light.cpp

```diff
@@ -237,7 +237,7 @@ void gl_shade_vertex(GLContext *c, GLVertex *v) {
 			d.Y = l->position.Y - v->ec.Y;
 			d.Z = l->position.Z - v->ec.Z;
 			dist = sqrt(d.X * d.X + d.Y * d.Y + d.Z * d.Z);
-			if (dist > 1E-3) {
+			if (dist != 1) {
```

Comment:
Also, if you're doing this test, you could also move the expensive `sqrt` inside, as there is no point finding the square root of unity.